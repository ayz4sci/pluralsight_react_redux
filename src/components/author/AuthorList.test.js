import React from 'react';
import renderer from 'react-test-renderer';
import AuthorList from './AuthorList';

describe ('AuthorList Test', () => {
    it('renders correctly using snapshot', () => {
        const props = {
            authors: [{id: 'A', firstName: 'B', lastName: 'C'}],
            deleteAuthor: () => Promise.resolve() 
        };
        const tree = renderer.create(<AuthorList {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
});