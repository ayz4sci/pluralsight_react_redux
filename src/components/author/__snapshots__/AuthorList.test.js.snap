// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorList Test renders correctly using snapshot 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th>
        First Name
      </th>
      <th>
        Last Name
      </th>
      <th>
         
      </th>
      <th>
         
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        <a
          onClick={[Function]}
          style={Object {}}
        >
          B
        </a>
      </td>
      <td>
        <a
          onClick={[Function]}
          style={Object {}}
        >
          C
        </a>
      </td>
      <td />
      <td />
      <td>
        <input
          className="btn btn-danger btn-sm"
          onClick={[Function]}
          type="submit"
          value="Delete"
        />
      </td>
    </tr>
  </tbody>
</table>
`;
